tests/src/test/java/org/eclipse/hono/tests/http/HttpTestBase.java

```diff
@@ -1103,6 +1114,18 @@ public void testUploadMessagesWithTtdThatReplyWithOneWayCommand(
                 });
     }
 
+    private Future<MultiMap> sendHttpRequestForGatewayOrDevice(final Buffer payload, final MultiMap requestHeaders,
+            final HttpCommandEndpointConfiguration endpointConfig, final String requestDeviceId) {
+        if (endpointConfig.isSubscribeAsGatewayForSingleDevice()) {
+            final String uri = getEndpointUri() + "/" + tenantId + "/" + requestDeviceId;
+            // GW uses PUT when acting on behalf of a device
+            return httpClient.update(uri, payload, requestHeaders, status -> status == HttpURLConnection.HTTP_OK);
```

Comment:
looks like you found the issue, too ;-) I had been investigating the problems with the test as well and also stumbled across this line ...

However,  I do not quite understand, why this fixes the issue. More specifically, I wonder, why during the test runs, the HTTP adapter actually seems to return a 202 to some of the requests. Do you know why that is?

Reply:
In the test runs, where 202 is returned, the creation of a command consumer fails because there is still a consumer from a previous test run which has not been fully closed yet. This causes a `ResourceConflictException` to be thrown and higher up caught in the HTTP adapter `createCommandConsumer` method:
https://github.com/eclipse/hono/blob/a170706410e9988a8a8471d6ef56ed5face2dcea/adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java#L1040-L1050

So, the `let the other request handle the command (if any)` is taken, the `responseReady` Future is triggered (with `null` as value which causes no `ttd` notification to be sent downstream!) and the response processing continues in the main `doUploadMessage` method, where this method
https://github.com/eclipse/hono/blob/a170706410e9988a8a8471d6ef56ed5face2dcea/adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java#L717
will set a 202 response because no command was received.

Reply:
When running the test where we expect one-way commands, we sometimes see re-tries of the request because the initial request yielded a 412. I erroneously assumed that the 412 was actually indicating exactly this case. However, the 412 (precondition failed) just indicates that the response contained an unexpected status code, in this case *not 200 but 202* ...
Ok, all seems to be in good shape now. Great work, @calohmn !!

Reply:
Yes, the fact that the original response status code is hidden and only a 412 is shown has caused a great deal of confusion with me as well.
https://github.com/eclipse/hono/blob/99b39c4c9fa4bc9bf99062e2b924c7e7df42d7af/tests/src/test/java/org/eclipse/hono/tests/CrudHttpClient.java#L598-L600
We should change this so that the original status code is somehow taken over (maybe just in the exception text). Then we can change the log output so that there isn't just the 412 here:
`re-trying request [43] which failed with unexpected status code 412`

Reply:
I agree, are you going to fix this?

Reply:
yes